test-framework/mongodb/src/main/java/io/quarkus/test/mongodb/MongoTestResource.java

```diff
@@ -24,15 +25,29 @@
 
 public class MongoTestResource implements QuarkusTestResourceLifecycleManager {
     public static final String PORT = "port";
+    public static final String VERSION = "version";
     private static final int DEFAULT_PORT = 27017;
     private static MongodExecutable MONGO;
 
     private static final Logger LOGGER = Logger.getLogger(MongoTestResource.class);
     private Integer port;
+    private IFeatureAwareVersion version;
 
     @Override
     public void init(Map<String, String> initArgs) {
         port = Optional.ofNullable(initArgs.get(PORT)).map(Integer::parseInt).orElse(DEFAULT_PORT);
+        version = Optional.ofNullable(initArgs.get(VERSION)).map(versionStr -> {
+            try {
+                return Version.valueOf(versionStr);
+            } catch (IllegalArgumentException e) {
+                try {
+                    return Version.Main.valueOf(versionStr);
+                } catch (IllegalArgumentException ex) {
+                    throw new IllegalArgumentException(
+                            String.format("Unable to convert %s to a known Mongo version", versionStr));
+                }
+            }
+        }).orElse(Version.Main.V4_0);
```

Comment:
I think we should move to 4.4 as the default, this is the last 4.x version and two new versions have already been published (5 and 6)

Reply:
Agreed, but that should probably be the subject of another PR.